app/server.js

```diff
@@ -1,31 +1,252 @@
+  };
+});
+
+// JWT_SECRET
+const JWT_SECRET = '9527e3a06a598251710743aa74e29e3681762684a01b184762469005a26afef3';
+
+// LOGIN
+app.post('/login', (req, res) => {
+  console.log('LOGIN...');
+  authHeader = req.headers.authorization;
+  console.log('AuthHeader: ', authHeader);
+  if(!authHeader || !authHeader.startsWith('Basic ')) {
+    return res.status(401).send({ error: 'Unauthorized' });
+  }
+  const base64Credentials = authHeader.split(' ')[1];
+  const credentials = Buffer.from(base64Credentials, 'base64').toString('utf-8');
+  const [username, password] = credentials.split(':');
+
+  // const { username, password } = req.body;
+  const user = users.find(user => user.login.username === username);
+  console.log('Username: ', username);
+  console.log('Password: ', password);
+
+  if (user &&  password === user.login.password) {
+    // GENERATES TOKEN
+    const token = jwt.sign({ username: user.login.username }, JWT_SECRET, {expiresIn: '1d' });
+
+    console.log('Token: ', token);
+
+    // ROUTE to (/{user.name.first})
+    const redirectUrl =`/${user.name.first}`;
+    console.log('ReDirectUrl: ', redirectUrl);
+
+    res.status(200).json({ token, redirectUrl: `/${user.name.first}`});
+  } else {
+    res.status(401).send('Username or password is incorrect');
+  }
+});
+
+// AUTHENTICATION MIDDLEWARE
+const authenticateJWT = (req, res, next) =>  {
+  let authHeader = req.headers['authorization'];
+  console.log('AUTHENTICATION');
+  console.log('AuthHeader: ', authHeader);
+
+  if (authHeader) {
+    const token = authHeader.split(' ')[1];
+    console.log('AuthHeader Deconstructed: ', token);
+
+    jwt.verify(token, JWT_SECRET, (err, user) => {
+      if (err) {
+        return res.status(403).json({ error: 'Invalid token' });
+      }
+
+      req.user = user;
+      next();
+    });
+  } else {
+    res.status(401).json({ error: 'Authorization header missing'  });
+  }
+};
+
+// AUTHENTICATED ROUTES
+app.get('/users', authenticateJWT, (req, res) => {
+  const userNames = users.map(user => user.name.first);
+  res.status(200).json({ users: userNames });
+});
+
+app.get('/:name', authenticateJWT, (req, res) => {
+  const userName = req.params.name.toLowerCase();
+  const user = users.find(user => user.name.first.toLowerCase() === userName);
+
+  if (user) {
+    const userCart = user.cart;
+    res.status(200).json({ userCart: userCart });
+  } else {
+    res.status(401).json({ error: 'Unauthorized' });
+  }; 
+});
+
+app.post('/:name', authenticateJWT, (req, res) => {
+  const userName = req.params.name.toLowerCase();
+  const user = users.find(user => user.name.first.toLowerCase() === userName);
+
+  if (!user) {
+    return res.status(401).json({ error: 'User not found.' });
+  };
+
+  if (!user.cart) {
+    user.cart = { items: [], total: 0 };
+  };
+
+  const newItem= {
+    product: req.body.product || 'glas',
+    quantity: req.body.quantity || 1,
+    price: req.body.price || 50,
+  };
```

Comment:
big no to send the whole product in the body from the client, you need to send the id and then just find this id in your products list in the BE. Also, why do you have a default? Big no as well, if the param is not provided, then you need to reject the call with 400 - client error.